daemon/prune.go

```diff
@@ -26,6 +26,7 @@ var (
 		"label":  true,
 		"label!": true,
 		"until":  true,
+		"dryRun": true,
```

Comment:
I would probably promote `dryRun` to a full option rather than a filter argument.

Reply:
@cpuguy83 when I remove this line, it is complaining me invalid parameters.

Reply:
This is probably because you are passing `dryRun` as a filter rather than a configuration option.
It looks like we don't currently accept any options on `prune` other than filter.

This would need to be added at the API level.

Reply:
@cpuguy83 can you give an example of any top level options. Like `--all`? Can you show me the file to add this.

Reply:
It will need to be decoded here: https://github.com/moby/moby/blob/f416e99a0574290f0fb65708b5f6a086182a14d0/api/server/router/container/container_routes.go#L713-L728

The backend interface definition will need to be updated here: 
https://github.com/moby/moby/blob/f416e99a0574290f0fb65708b5f6a086182a14d0/api/server/router/container/backend.go#L67

The options will need to be encoded here: https://github.com/moby/moby/blob/f416e99a0574290f0fb65708b5f6a086182a14d0/client/container_prune.go#L13-L36